### PR TITLE
Trim SSE event names.

### PR DESCRIPTION
**Pull request by mcdee**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

When sending SSE events the event name contains trailing whitespace.  This patch removes the whitespace before processing the event, ensuring it is easily parsed by clients.

### CODE REVIEW

File: beacon-chain/rpc/apimiddleware/custom_handlers_test.go [hunk 1]
```diff
@@ -202,7 +202,7 @@ func TestWriteEvent(t *testing.T) {
 	require.NoError(t, err)
 	msg := &sse.Event{
 		Data:  bData,
-		Event: []byte("test_event"),
+		Event: []byte("test_event "),
```
Comment (rkapka) [inline beacon-chain/rpc/apimiddleware/custom_handlers_test.go:205]: Please a separate test named something like `TestWriteEvent_WhitespaceRemoved`.

**Review by rkapka — CHANGES_REQUESTED** [review]
Thanks for the PR. I have one small request.

File: beacon-chain/rpc/apimiddleware/custom_handlers_test.go (hunk 1 above)
Comment (mcdee) [reply beacon-chain/rpc/apimiddleware/custom_handlers_test.go:227]: Updated.

**Comment by rkapka** [comment]
For some reason the new test does not pass.

**Comment by mcdee** [comment]
I took another look and changed the test to use `receiveEvents` rather than `writeEvent` as the base.  `receiveEvents` is the only place that calls `writeEvent`, and the logic to trim the event is up in `receiveEvents`so that's where it makes sense to inject the event with the trailing space.

**Comment by rauljordan** [comment]
PTAL @rkapka

**Review by rkapka — APPROVED** [review]

**Pull request merged**